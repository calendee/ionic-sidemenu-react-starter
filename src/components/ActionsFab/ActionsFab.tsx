import React, { useContext } from 'react';
import {
  IonBadge,
  IonFab,
  IonFabButton,
  IonFabList,
  IonIcon,
} from '@ionic/react';

import Modal from '../Modal/Modal';
import useToggle from '../../hooks/useToggle';

import { EventsContext } from 'providers/Events/State';
import { ActionTypes } from 'providers/Events/actions';
import EventsList from '../EventsList/EventsList';

const ActionsFab: React.FC = () => {
  const { state, dispatch } = useContext(EventsContext);
  const [open, setOpen] = useToggle(false);

  const resetCounter = () => {
    dispatch({
      type: ActionTypes.RESET_EVENTS,
      payload: null,
    });
  };

  return (
    <>
      <IonFab vertical="bottom" horizontal="end" slot="fixed">
        <IonFabButton>
          <IonBadge color="primary">{state.count}</IonBadge>
        </IonFabButton>
        <IonFabList side="top">
          <IonFabButton color="danger" onClick={setOpen}>
            <IonIcon icon="list" />
          </IonFabButton>
          <IonFabButton color="danger" onClick={resetCounter}>
            <IonIcon icon="close" />
          </IonFabButton>
        </IonFabList>
      </IonFab>
      <Modal close={setOpen} open={open} title="Events List">
        <EventsList />
      </Modal>
    </>
  );
};

export default ActionsFab;
