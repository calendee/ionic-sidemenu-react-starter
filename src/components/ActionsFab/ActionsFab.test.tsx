import React from 'react';

import { fireEvent, renderWithUserAndEvents } from '../../../test/test-utils';
import ActionsFab from './ActionsFab';

describe('ActionsFab', () => {
  test('renders and matches snapshot', async () => {
    const { container, getByText } = renderWithUserAndEvents(<ActionsFab />);
    getByText('0');
    fireEvent.click(container);
    expect(container).toMatchSnapshot();
  });
});
