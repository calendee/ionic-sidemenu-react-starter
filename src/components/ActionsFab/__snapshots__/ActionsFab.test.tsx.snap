// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ActionsFab renders and matches snapshot 1`] = `
<div>
  <ion-app>
    <span
      data-testid="actions-fab"
    >
      <ion-fab
        horizontal="end"
        slot="fixed"
        vertical="bottom"
      >
        <ion-fab-button>
          <ion-badge
            color="primary"
          >
            0
          </ion-badge>
        </ion-fab-button>
        <ion-fab-list
          side="top"
        >
          <ion-fab-button
            color="danger"
          >
            <ion-icon
              icon="list"
            />
          </ion-fab-button>
          <ion-fab-button
            color="danger"
          >
            <ion-icon
              icon="close"
            />
          </ion-fab-button>
        </ion-fab-list>
      </ion-fab>
    </span>
  </ion-app>
</div>
`;
